container/container.go

```diff
@@ -174,115 +154,115 @@ func (c *Container) Start(srv *service.Service, instanceHash, runnerHash, instan
 		if err != nil {
 			return err
 		}
-		stopGracePeriod := defaultStopGracePeriod
-		maxAttempts := defaultMaxAttempts
-		specs = append(specs, swarm.ServiceSpec{
-			Annotations: swarm.Annotations{
-				Name: depName,
-				Labels: map[string]string{
-					"mesg.engine":                c.engineName,
-					"mesg.service":               srv.Hash.String(),
-					"mesg.instance":              instanceHash.String(),
-					"mesg.runner":                runnerHash.String(),
-					"mesg.dependency":            dep.Key,
-					"com.docker.stack.namespace": depName,
-				},
-			},
-			TaskTemplate: swarm.TaskSpec{
-				ContainerSpec: &swarm.ContainerSpec{
-					Image: dep.Image,
-					Labels: map[string]string{
-						"com.docker.stack.namespace": depName,
-					},
-					Env:             dep.Env,
-					Args:            dep.Args,
-					Command:         strings.Fields(dep.Command),
-					Mounts:          append(volumes, volumesFrom...),
-					StopGracePeriod: &stopGracePeriod,
-				},
-				Networks: []swarm.NetworkAttachmentConfig{
-					{
-						Target:  networkID,
-						Aliases: []string{dep.Key},
-					},
-				},
-				RestartPolicy: &swarm.RestartPolicy{
-					Condition:   swarm.RestartPolicyConditionOnFailure,
-					MaxAttempts: &maxAttempts,
-				},
+		exposedPort, portBindings, err := nat.ParsePortSpecs(dep.Ports)
+		containerConfig := &containertypes.Config{
+			Image: dep.Image,
+			Labels: map[string]string{
+				"mesg.engine":     c.engineName,
+				"mesg.service":    srv.Hash.String(),
+				"mesg.instance":   instanceHash.String(),
+				"mesg.runner":     runnerHash.String(),
+				"mesg.dependency": dep.Key,
 			},
-			EndpointSpec: &swarm.EndpointSpec{
-				Ports: convertPorts(dep.Ports),
+			Env:          dep.Env,
+			ExposedPorts: exposedPort,
+		}
+		hostConfig := &containertypes.HostConfig{
+			PortBindings: portBindings,
+			Mounts:       append(volumes, volumesFrom...),
+			RestartPolicy: containertypes.RestartPolicy{
```

Comment:
Not critical for this PR but I would be strict on the restart policy here as it is used only for e2e and not allow any restart